gno.land/pkg/gnoweb/markdown/ext_column.go

```diff
@@ -0,0 +1,313 @@
+package markdown
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/parser"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/text"
+	"github.com/yuin/goldmark/util"
+)
+
+// ErrInvalidColumnFormat indicates an invalid columns format.
+var ErrInvalidColumnFormat = errors.New("invalid columns format")
+
+// Define NodeKind for Column
+var KindColumn = ast.NewNodeKind("Column")
+
+type ColumnTag int
+
+const (
+	ColumnTagUndefined ColumnTag = iota
+	ColumnTagOpen
+	ColumnTagSep
+	ColumnTagClose
+)
+
+var columnTagNames = map[ColumnTag]string{
+	ColumnTagUndefined: "ColumnTagUndefined",
+	ColumnTagOpen:      "ColumnTagOpen",
+	ColumnTagSep:       "ColumnTagSep",
+	ColumnTagClose:     "ColumnTagClose",
+}
+
+// ColumnNode represents a semantic tree for "column".
+type ColumnNode struct {
+	ast.BaseBlock
+	Index int
+	Tag   ColumnTag
+	ctx   *columnContext // context between blocks is unique.
+}
+
+// Dump implements Node.Dump.
+func (n *ColumnNode) Dump(source []byte, level int) {
+	kv := map[string]string{
+		"tag":      columnTagNames[n.Tag],
+		"head_ref": strconv.Itoa(n.ctx.RefHeadingLevel),
+	}
+	if n.Tag == ColumnTagSep {
+		kv["index"] = strconv.Itoa(n.Index)
+	}
+	if n.ctx.Error != nil {
+		kv["error"] = n.ctx.Error.Error()
+	}
+
+	ast.DumpHelper(n, source, level, kv, nil)
+}
+
+// Kind implements Node.Kind.
+func (*ColumnNode) Kind() ast.NodeKind {
+	return KindColumn
+}
+
+// NewColumn initializes a ColumnNode object.
+func NewColumn(ctx *columnContext, index int, tag ColumnTag) *ColumnNode {
+	return &ColumnNode{ctx: ctx, Index: index, Tag: tag}
+}
+
+// columnParser struct and its methods are used for parsing columns.
+type columnParser struct{}
+
+// Trigger returns the trigger characters for the parser.
+func (*columnParser) Trigger() []byte {
+	return []byte{':', '<', '#'}
+}
+
+var columnContextKey = parser.NewContextKey()
+
+// columnContext struct and its methods are used for handling column context.
+type columnContext struct {
+	IsOpen          bool
+	Error           error
+	Index           int
+	RefHeadingLevel int // serves as a level reference for separators
+}
+
+func isPreviousNodeTag(node ast.Node, tag ColumnTag) bool {
```

Comment:
deadcode (and in general there are a lot of hard to read code) can you make a quick cleanup?